web_client/src/scenes/Landing/index.js

```diff
@@ -0,0 +1,23 @@
+import React, { Component } from 'react';
+import { connect} from 'react-redux';
+import { Redirect } from 'react-router';
+import Landing from './Landing';
+import { getAccessToken } from 'selectors';
+import { redirectToAuthorizationPage } from 'services/api';
+
+class LandingContainer extends Component {
+  render = () => (
+    this.props.accessToken
+      ? <Redirect to='/playlists'/>
+      : <Landing onLoginButtonClicked={e => {e.preventDefault(); redirectToAuthorizationPage();}} />
```

Comment:
I think you can put this logic directly int the routes.js file. But it's fine here too. Why not have the `e.preventDefault()` logic in the Landing component?

is this useful?
https://reacttraining.com/react-router/web/example/auth-workflow

Reply:
so this isn't routing to my authorization page. this routes to the spotify authorization page (where the user logs in via spotify's portal and then is redirected to playlistsouffle.com. so to me it's more part of the spotify 'api' than my internal routing.

what `routes.js` file are you talking about?

yeah putting `e.preventDefault` into the `Landing` component makes sense.

Reply:
i will look into this. clearly the routing / auth / loading needs the most refactoring (which i agree with, since i was really confused while i worked it out)